exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/HashJoinBatch.java

```diff
@@ -381,16 +409,14 @@ public HashJoinMemoryCalculator getCalculatorImpl() {
 
   @Override
   public IterOutcome innerNext() {
-    if (!prefetched) {
+    if (!prefetchedBuild) {
       // If we didn't retrieve our first data hold batch, we need to do it now.
-      prefetched = true;
-      prefetchFirstBatchFromBothSides();
+      prefetchedBuild = true;
+      prefetchFirstBuildBatch();
 
       // Handle emitting the correct outcome for termination conditions
-      // Use the state set by prefetchFirstBatchFromBothSides to emit the correct termination outcome.
+      // Use the state set by prefetchFirstBuildBatch to emit the correct termination outcome.
```

Comment:
Code cleaning: The check below (switch statement) can be done as part of the identical check after executeBuildPhase() (which is skipped anyway if STOP).  Also maybe move the "wasKilled" check first thing in innerNext().